src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/LocalDateDeserializer.java

```diff
@@ -75,6 +76,11 @@ public LocalDate deserialize(JsonParser parser, DeserializationContext context)
                 if(string.length() == 0) {
                     return null;
                 }
+
+                if(string.contains("T")) {
+                    return LocalDate.parse(string, DateTimeFormatter.ISO_LOCAL_DATE_TIME);
```

Comment:
It could be just me, but I think leniency should have an off switch. With this change it is impossible to force failure in case the data is not exactly in the expected format.